File: práctica 2/Circulo.py
import math
import turtle

class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y
        
    def __str__(self):
        return f"({self.x}, {self.y})"
        
class Circulo:
    def __init__(self, c, r):
        self.c = c
        self.r = r
        
    def __str__(self):
        return f"Circulo con centro en {self.c} y radio {self.r}"
        
    def dibuja(self):
        print(f"Dibujando circulo con centro en {self.c} y radio {self.r}")

        turtle.penup() 
        turtle.goto(self.c.x, self.c.y - self.r)  
        turtle.pendown()  
        
        turtle.circle(self.r)  
    
    def area(self):
        return math.pi * (self.r ** 2)
        
    def perimetro(self):
        return 2 * math.pi * self.r

centro = Punto(0, 0) 
circulo = Circulo(centro, 170)  

print(circulo)

circulo.dibuja()

print(f"Área: {circulo.area():.2f}")
print(f"Perímetro: {circulo.perimetro():.2f}")

turtle.done()

File: práctica 2/Circulo.java
public class Main {

    static class Punto {
        int x, y;
        Punto(int x, int y) { this.x = x; this.y = y; }
        public String toString() { return "(" + x + ", " + y + ")"; }
    }

    static class Circulo {
        Punto c;
        double r;
        Circulo(Punto c, double r) { this.c = c; this.r = r; }
        public String toString() { return "Circulo en " + c + " con radio " + r; }
        public double area() { return Math.PI * r * r; }
        public double perimetro() { return 2 * Math.PI * r; }
    }

    public static void main(String[] args) {
        Circulo circulo = new Circulo(new Punto(7, 9), 11);
        System.out.println(circulo);
        System.out.printf("Área: %.2f\nPerímetro: %.2f\n", circulo.area(), circulo.perimetro());
    }
}
